brainiak/isfc.py

```diff
@@ -0,0 +1,148 @@
+#  Copyright 2017 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Intersubject analyses (ISC/ISFC)
+
+Functions for computing intersubject correlation (ISC) and intersubject
+functional correlation (ISFC), and utility functions for loading subject data
+files and masks
+"""
+
+# Authors: Christopher Baldassano and Mor Regev
+# Princeton University, 2017
+
+from brainiak.fcma.util import compute_correlation
+from scipy import stats
+import nibabel as nib
+import numpy as np
+
+
+def isc(D, collapse_subj=True):
+    """Intersubject correlation
+
+    For each voxel, computes the correlation of each subject's timecourse with
+    the mean of all other subjects' timecourses. By default the result is
+    averaged across subjects, unless collapse_subj is set to False.
+
+    Parameters
+    ----------
+    D : voxel by time by subject ndarray
+        fMRI data for which to compute ISC
+
+    collapse_subj : bool, default:True
+        Whether to average across subjects before returning result
+
+    Returns
+    -------
+    ISC : voxel ndarray (or voxel by subject ndarray, if collapse_subj=False)
+        pearson correlation for each voxel, across subjects
+    """
+
+    n_vox = D.shape[0]
+    n_subj = D.shape[2]
+    ISC = np.zeros((n_vox, n_subj))
+
+    for loo_subj in range(n_subj):
+        group = np.mean(D[:, :, np.arange(n_subj) != loo_subj], axis=2)
+        subj = D[:, :, loo_subj]
+        for v in range(n_vox):
+            ISC[v, loo_subj] = stats.pearsonr(group[v, :], subj[v, :])[0]

```

Comment:
Why don't you use `fcma.util.compute_correlation` here as well? If it is because of the data layouts, you can simply convert them accordingly. I believe even adding the layout conversion time up, the overall running time should outperform `scipy.stats.pearsonr`.